### PR TITLE
MINOR: Refactor unit tests around RocksDBConfigSetter

### PR DESCRIPTION
**Pull request by guozhangwang**
1. Extract the mock RocksDBConfigSetter into a separate class.
2. De-dup unit tests covering RocksDBConfigSetter.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java [hunk 1]
```diff
@@ -297,9 +298,21 @@ public void shouldNotThrowExceptionOnRestoreWhenThereIsPreExistingRocksDbFiles()
     public void shouldCallRocksDbConfigSetter() {
         MockRocksDbConfigSetter.called = false;
 
+        final Properties props = StreamsTestUtils.getStreamsConfig();
```
Comment (guozhangwang) [inline streams/src/test/java/org/apache/kafka/streams/state/internals/RocksDBStoreTest.java:302]: This is to illustrate the usage of passing in any user-customized configurable value or shared objects to the setter. @rodesai @brianbushree

**Comment by guozhangwang** [comment]
I created this PR mainly to illustrate that one can pass in shared object or configurable values to the setter as well.

**Comment by guozhangwang** [comment]
test this.

**Review by abbccdda — APPROVED** [review]
LGTM!

**Comment by guozhangwang** [comment]
Thanks @abbccdda for the reviews!

**Pull request merged**